test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlWriterTests.cs

```diff
@@ -1203,6 +1203,21 @@ public void CanWriteUrlEscapeFunction()
             Assert.Equal(expected, csdlStr);
         }
 
+        [Theory]
+        [InlineData("4.0")]
+        [InlineData("4.01")]
+        public void ValidateEdmxVersions(string odataVersion)
+        {
+            string xml = "<?xml version=\"1.0\" encoding=\"utf-16\"?><edmx:Edmx Version=\"" + odataVersion + "\" xmlns:edmx=\"http://docs.oasis-open.org/odata/ns/edmx\"><edmx:DataServices /></edmx:Edmx>";
+
+            // Specify the model
+            EdmModel edmModel = new EdmModel(false);
+            edmModel.SetEdmVersion(odataVersion == "4.0" ? EdmConstants.EdmVersion4 : EdmConstants.EdmVersion401);

```

Comment:
>EdmConstants.EdmVersion4 [](start = 59, length = 24)

same as, can move it to [InlineData('")]

Reply:
as above, I like that a lot better, but I got an error from the test framework when I tried to put EdmConstants.EdmVersion4 in the inline data.

---
In reply to: [290414340](https://github.com/OData/odata.net/pull/1476#discussion_r290414340) [](ancestors = 290414340)

Reply:
Straight. It should work for Enum. Same as:
```        
[Theory]
[InlineData(EdmTypeKind.Complex)]
[InlineData(EdmTypeKind.Primitive)]
public void OperationReturnTypeWithCollectionOfAbstractTypeShouldError(EdmTypeKind typeKind)
```